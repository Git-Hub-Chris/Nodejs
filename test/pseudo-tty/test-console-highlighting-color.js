'use strict';
require('../common');
const assert = require('assert');
const { Console } = require('console');

// Work with 16 colors
process.env.FORCE_COLOR = 1;
delete process.env.NODE_DISABLE_COLORS;
delete process.env.NO_COLOR;

// Arguments validations
const invalidTypeError = {
  code: 'ERR_INVALID_ARG_TYPE',
  name: 'TypeError',
  message: /must be of type/,
};
const mustBeOneOfError = {
  code: 'ERR_INVALID_ARG_VALUE',
  name: 'TypeError',
  message: /must be one of/,
};
const options = {
  stdout: process.stdout,
  stderr: process.stderr,
  ignoreErrors: true,
};

options.highlight = { warn: 'abc' };
assert.throws(() => { new Console(options); }, invalidTypeError);

options.highlight = { warn: { bgColor: 'abc' } };
assert.throws(() => { new Console(options); }, mustBeOneOfError);

options.highlight = { warn: { indicator: null } };
assert.throws(() => { new Console(options); }, invalidTypeError);

options.highlight = { error: 'abc' };
assert.throws(() => { new Console(options); }, invalidTypeError);

options.highlight = { error: { bgColor: 'abc' } };
assert.throws(() => { new Console(options); }, mustBeOneOfError);

options.highlight = { error: { indicator: null } };
assert.throws(() => { new Console(options); }, invalidTypeError);

options.highlight = {
  warn: { bgColor: 'bgYellow', indicator: '[W]' },
  error: { bgColor: 'bgRed', indicator: '[E]' },
};

const newInstance = new Console(options);
newInstance.warn(new Error('test'));
newInstance.error(new Error('test'));
