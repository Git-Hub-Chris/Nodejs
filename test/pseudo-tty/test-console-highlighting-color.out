*[43m[W] Error: test*[K
    at * (*test-console-highlighting-color.js:*)*[K
*    at *[K
*    at *[K
*    at *[K
*    at *[K
*    at *[K
*    at *[K*[49m

*[41m[E] Error: test*[K
    at * (*test-console-highlighting-color.js:*)*[K
*    at *[K
*    at *[K
*    at *[K
*    at *[K
*    at *[K
*    at *[K*[49m
